RIOT-OS/RIOT
# tests/devfs: move tests-devfs out of unittests

### Contribution description

This moves unittests/tests-devfs to its own test.
It prevents requiring `periph_hwrng` in the `unittests`.


### Testing procedure

I tried compiling the test with `buildtest` and `BUILD_IN_DOCKER` and all supported boards compiled. This will be tested by CI.

The test also succeeds on both `native` and `nrf52dk` locally, and should also in CI.


<details><summary><code>RIOT_CI_BUILD=1 BOARD=nrf52dk make --no-print-directory -C tests/devfs/ flash test</code></summary>

```
RIOT_CI_BUILD=1 BOARD=nrf52dk make --no-print-directory -C tests/devfs/ flash test
Building application "tests_devfs" for "nrf52dk" with MCU "nrf52".

   text    data     bss     dec     hex filename       
  13536     244    3012   16792    4198 /home/harter/work/git/RIOT/tests/devfs/bin/nrf52dk/tests_devfs.elf
/home/harter/work/git/RIOT/dist/tools/jlink/jlink.sh flash /home/harter/work/git/RIOT/tests/devfs/bin/nrf52dk/tests_devfs.bin
### Flashing Target ###
### Flashing at base address 0x0 with offset 0 ###
SEGGER J-Link Commander V6.42d (Compiled Feb 15 2019 13:56:53)
DLL version V6.42d, compiled Feb 15 2019 13:56:43
...

Script processing completed.

/home/harter/work/git/RIOT/dist/tools/pyterm/pyterm -p "/dev/ttyACM0" -b "115200"
Twisted not available, please install it if you want to use pyterm's JSON capabilities
2019-07-17 14:45:30,462 - INFO # Connect to serial port /dev/ttyACM0
Welcome to pyterm!
Type '/exit' to exit.
2019-07-17 14:45:33,817 - INFO # main(): This is RIOT! (Version: buildtest)
2019-07-17 14:45:33,818 - INFO # ....
2019-07-17 14:45:33,819 - INFO # OK (4 tests)
```
</details>

The `iotlab-m3` can again compile and succeed with the `tests/unittests`.

<details><summary><code>RIOT_CI_BUILD=1 BOARD=iotlab-m3 make -C tests/unittests/ flash test</code></summary>

```
RIOT_CI_BUILD=1 BOARD=iotlab-m3 make -C tests/unittests/ flash test
Twisted not available, please install it if you want to use pyterm's JSON capabilities
2019-07-17 14:41:35,364 - INFO # Connect to serial port /dev/ttyUSB1
Welcome to pyterm!
Type '/exit' to exit.
2019-07-17 14:41:38,741 - INFO # �main(): This is RIOT! (Version: buildtest)
2019-07-17 14:42:39,903 - INFO # ...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
2019-07-17 14:42:39,904 - INFO # OK (943 tests)
```
</details>

### Issues/PRs references

`unittests` not compiling on `iotlab-m3` was found during Release testing 2019.07-RC1
https://github.com/RIOT-OS/Release-Specs/issues/128#issuecomment-511451764

Splitting the test out was mentioned in the introduction PR https://github.com/RIOT-OS/RIOT/pull/7421#pullrequestreview-186105492
